app/models/completed_lesson.rb

```diff
@@ -1,4 +1,10 @@
 class CompletedLesson < ApplicationRecord
 	belongs_to :user
 	belongs_to :lesson
+
+	private
+
+	def set_default_to_completed
+		self.completed ||= true
+	end
```

Comment:
This code is not being called anywhere. If you want to run it during the validation step, you can call it using a callback

Documentation: https://guides.rubyonrails.org/active_record_callbacks.html#available-callbacks

Reply:
Only `nil` values should change to `true`. This code would change `completed: false` to `completed: true`

```ruby
irb(main):004:0> a = false
=> false
irb(main):005:0> a ||= true
=> true
irb(main):006:0> a
=> true
```

`||=` affects any falsy value (`nil` and `false`), so your code should be a bit different in order to only affect `nil`.
A TDD approach here would help a lot. You can write a test to ensure that `completed: false` remains `false`.

Reply:
I changed my code to only set a default if the value is nil. I tested it with a TDD approach, and it seemed to work.  Can you please check to verify that it is now correct?  Thank you.